Source/UI/Windows/SpeakerSetup/SpeakerTreeComponent.cpp

```diff
@@ -532,4 +560,21 @@ void SpeakerComponent::setupHighPass()
     addAndMakeVisible(highpass);
 }
 
+void SpeakerComponent::labelTextChanged (juce::Label* label)
+{
+    if (label != &id)
+        return;
+
+    auto const currentId = id.getText ().getIntValue ();
+    auto const clampedId { std::clamp (currentId, 1, MAX_NUM_SPEAKERS) };
+    if (clampedId != currentId)
+        id.setText (juce::String (clampedId), juce::dontSendNotification);
+
+    // TODO: for now speaker ID edition isn't undoable; it's impossible to track the proper previous and next
+    // output patch numbers when lining up multiple undos/redos. To do this properly, we should change
+    // MainContentComponent::speakerOutputPatchChanged() and all related logic to use speaker UUIDs instead of
```

Comment:
I think this should be opened as a separate issue so that we do not forget about it

Reply:
just done!